### ISSUE TITLE
Bot does not detect wishlist characters correctly

### ISSUE THREAD

**Issue by KingJackFrost** [chiyadev/MudaeFarm#151]
Thanks for making this bot, first of all. It's been a great learning tool while I study C#, and it's also cool to see it in action!

I noticed that the bot looks for specific characters with their series, but also uses it for exact character matching. For example, "Naruko Anjou (Anohana: The Flower We Saw That Day)" will not match "Naruko Anjou." I assume that this bot will work when there are multiple alias, and Mudae tags it as the character from that show, however. Below is a better example of what it looks like in the console output.

![image](https://user-images.githubusercontent.com/39563259/83777120-a7237480-a64e-11ea-8e16-789e7f68de57.png)

I wanted to bring this issue to your attention. Ways to circumvent this issue I assume would be to manually add characters to your wishlist. In the above example, simply type "Naruko Anjou" into the wished-characters, and it should work, but I haven't tested the wish-characters functionality fully yet.

**Comment by phosphene47** [comment]
This behavior is intentional, and is already explained in detail [here](https://github.com/chiyadev/MudaeFarm#wished-characters):

> If a character name contains information in brackets for disambiguation, this must be included.

You need to add exactly `Naruko Anjou (Anohana: The Flower We Saw That Day)` including the brackets to the wishlist. You could also add `Naruko Anjou*` where `*` acts as a wildcard. You could even use the JSON syntax which supports regex.

Why is this so? Because information in the brackets is used for disambiguation. Although unlikely, there could be another character named Naruko Anjou from a different anime.

**Comment by phosphene47** [comment]
As I said above, if you wish to claim `Naruko Anjou (Anohana: The Flower We Saw That Day)` you need to add *exactly that*, including the brackets, to MudaeFarm's wishlist channel. But I think I'm misunderstanding what you mean.

Why not use `Naruko Anjou*` instead? It will match both `Naruko Anjou (Anohana: The Flower We Saw That Day)` and `Naruko Anjou`.

**Comment by KingJackFrost** [comment]
For your first comment, I'll try to clarify. 

I understand that you need to add `Naruko Anjou (Anohana: The Flower We Saw That Day)` to prevent disambiguation. However, let's say that there is only one Naruko Anjou. What this means is that there no disambiguation; should someone use a roll to roll Naruko Anjou, Mudae will show `Naruko Anjou.` Now, let's say you add your wishlist to the MudaeFarm via the channel wished-users. In your wishlist, you have `Naruko Anjou.` When `Naruko Anjou` pops up, Mudae does not detect it correctly. As seen in the screenshot, Mudae reads from the wishlist the line,  `Naruko Anjou (Anohana: The Flower We Saw That Day)' and not `Naruko Anjou' (the correct interpretation). As a result, Mudae misses this roll and does not claim for the user.

Thus, I said to circumvent this issue, is to add both the wishlist to the wishlist-user and also add a separate character `Naruko Anjou` to the wished-characters. Does that make sense? I could try to explain it another way.

For the embed, do you mean the data for Naruko Anjou? I am confused here.

**Comment by phosphene47** [comment]
I understand the situation now, but you are misinformed about how the #wishlist-users channel works.

When your Mudae wishlist is public and someone rolls a character that you wished in your Mudae wishlist, Mudae will ping you in the format, `Wished by @KingJackFrost, @user, @user...`. MudaeFarm detects these pings. It never *reads* your Mudae wishlist in the first place. This is the reason why your Mudae wishlist needs to be public; otherwise Mudae won't ping you.

**Comment by KingJackFrost** [comment]
Sure. I'm not sure if this would help, so I tried to screenshot what I assumed you meant by the embed message. It appears to be a default "did not wish" message.

![image](https://user-images.githubusercontent.com/39563259/83782416-65e29300-a655-11ea-8e52-bb2f9ac79957.png)

**Comment by KingJackFrost** [comment]
(Had to delete the previous comment, had some sensitive information)

Here is the roll message from Mudae farm, if it helps.

![image](https://user-images.githubusercontent.com/39563259/83783122-a93d0180-a655-11ea-83aa-1835d50e680d.png)

**Comment by KingJackFrost** [comment]
(Had to delete the previous comment due to sensitive information).

Here is the roll message I assumed u meant.

![image](https://user-images.githubusercontent.com/39563259/83783467-c4a80c80-a655-11ea-8c74-584a98624cfb.png)

**Comment by phosphene47** [comment]
How about a screenshot of the roll *on Discord*? I'm talking about the message Mudae sends you after $w (character name, anime, image), not the log.

**Comment by KingJackFrost** [comment]
I'd have to test it again. I used Naruko Anjou as an example, but I noticed this occurred specifically for when I wished Rias Gremory. Here is what it looks like for that character.

![image](https://user-images.githubusercontent.com/39563259/83785028-531c8e00-a656-11ea-8664-7ff70d7422bc.png)

Some things that might not be discernable from the picture:
1. The message Mudae sends is two separate messages: One from the @wish part, one for the actual character embed.
2. I had to manually claim after I realized the bot didn't claim the character (if you're wondering why there is a react)

**Comment by phosphene47** [comment]
Thanks for the screenshot!

> The message Mudae sends is two separate messages: One from the @wish part, one for the actual character embed.

I made an assumption that the ping and the embed are one message (or Mudae changed this without me knowing.) It seems like this caused the character to not be detected.

**Comment by phosphene47** [comment]
Yes, that workaround should work. `Wished by @user` has no effect on MudaeFarm's own wishlist. Also, https://github.com/chiyadev/MudaeFarm/issues/133#issuecomment-636549741

**Comment by KingJackFrost** [comment]
Thanks, that makes my life easier (I've been putting messages line by line). Keeping this issue open until it's fixed for your convenience.
